text/0249-v2-experiments.md

```diff
@@ -638,8 +638,71 @@ use a naming scheme that convey its non-final nature for the following reasons:
    that this is not the final version of the API, and its deprecation is
    expected.
 
+### Modules lifecycle
+
+This section discuss adding a new modules to `aws-cdk-lib`.
+
+#### Alpha (optional)
+
+This stage is intended for modules that are in early development stage, they
+change rapidly and may incurs many breaking changes. These modules will be
+released separately from `aws-cdk-lib` and will have their own version, which
+allows for breaking changes in accordance to semver, e.g `0.x` or prerelease
+qualifiers.
+
+In order to release a module as alpha, the following conditions must be met:
+
+1. Module from `aws-cdk-lib` **can not** depend on it.
+2. Can not depend on any other alpha module<sup>\*\*</sup>.
+
+The purpose of the first condition is to prevent cyclic dependencies. To
+illustrate the purpose of the second condition lets look at an example. Assume
+we have an alpha module A, which depends on an alpha module B, graduating module
+A means it will be added to `aws-cdk-lib`, since module A depends on module B,
+it means that adding module A to `aws-cdk-lib` breaks condition 1 and creates a
+cyclic dependency. To prevent the cycle we will have to graduate B as well,
+which is not necessarily what we want. Additionally, if module B is an alpha
+module, module A will have to declare a fixed dependency on it, locking its
+consumers to the same version of module B - defeating the purpose of
+peerDependencies.
+
+> \*\* _There might be exceptions that will be considered on a per case basis_
+
```

Comment:
So if an external contributor wants to contribute L2 for a module that we only have CFN classes for, what is their process? It seems like they won't be able to just submit a PR against the CDK repo. Someone will need to create a repo for them, etc. Worth mentioning at least, and acknowledging that this is not as easy as today.

Reply:
We may want to consider third party ownership here. Allowing contributors to make and maintain their own repositories, provided we give them the templates/tools that make it appropriate for eventual inclusion in aws-cdk-lib, may be beneficial. This would standardize a path for contributing wholly new functionality that can be leveraged by users without the core team getting in the way. If something is ultimately deemed to not be relevant for inclusion in aws-cdk-lib it can just continue to be released as a separate module.

This has the downside of not being published under the aws-cdk namespaces in package managers and discoverability may be an issue. Additionally even with tools to make this as easy as possible contributing to an existing repository will always be easier.

Reply:
> Someone will need to create a repo for them, etc. Worth mentioning at least, and acknowledging that this is not as easy as today.

We will have a single monorepo with all experimental modules versioned independently